### PR TITLE
Front logic

### PR DESCRIPTION
**Pull request by mdebowska**
Changed way to set the winner - now is possible to disable winner and check both as draw.
Made BetCreatorForm as a separated component from BetListPage.

### CODE REVIEW

File: src/components/BetCreatorForm.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    try {
+      const response = await betConsumer.addNewBet(request)
+      onBetAdded(response)
+    } catch (e: any) {
+      console.error('failed')
+    }
+  }
+  return (
+    <div className="form flex space-between">
+      <textarea
```
Comment (radek2s) [inline src/components/BetCreatorForm.tsx:25]: We can use TextField from [FluentUI library](https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield)

File: src/components/BetCreatorForm.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        placeholder="Enter a bet title"
+        id="bet-title"
+        onChange={(e) => setRequest({ ...request, title: e.target.value })}></textarea>
```
Comment (radek2s) [inline src/components/BetCreatorForm.tsx:28]: I think that we should change our aproach to input updates
This guy explain how to change from rendering whole component every time we type any letter into textarea. [Tutorial](https://youtu.be/GGo3MVBFr1A?t=72).

I think that is good ocassion to check it in real life example.

**Review by radek2s — CHANGES_REQUESTED** [review]

File: src/models/BetEntry.ts [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
+
 export default class BetEntry {
   id: number | string
   title: string
   description: string
   option1: string
   option2: string
-  isFinished: boolean
-  winner: boolean
+  winner: Winner
```
[the remark this answers is not in the document]
Comment (mdebowska) [reply src/models/BetEntry.ts]: changet to BetResolve and Pending

File: src/models/BetEntry.ts [hunk 4]
```diff
@@ -1,27 +1,31 @@
+export enum Winner {
```
[the remark this answers is not in the document]
Comment (mdebowska) [reply src/models/BetEntry.ts]: Changed to BetResolve

File: src/components/BetElement.tsx [hunk 5]
```diff
@@ -9,9 +9,9 @@ interface IBetElement {
 const BetElement: React.FC<IBetElement> = ({ bet, betDelete, betUpdate }) => {
   const [request, setRequest] = React.useState<BetEntry>({ ...bet })
 
-  const updateBet = (isFinished: boolean, winner: boolean) => {
-    setRequest({ ...request, isFinished, winner })
-    betUpdate({ ...request, isFinished, winner }) //TODO: możliwość zaznaczania i odznaczania obu zakładów
+  const updateBet = (winner: Winner) => {
+    setRequest({ ...request, winner })
+    betUpdate({ ...request, winner })
   }
```
[the remark this answers is not in the document]
Comment (mdebowska) [reply src/components/BetElement.tsx]: changed

File: src/components/BetElement.tsx [hunk 6]
```diff
@@ -43,33 +43,59 @@ const BetElement: React.FC<IBetElement> = ({ bet, betDelete, betUpdate }) => {
...
+                  break
+                case Winner.Person2:
+                  updateBet(Winner.Draw)
+                  break
+                case Winner.Draw:
+                  updateBet(Winner.Person2)
+                  break
+              }
+            }}
           />
```
[the remark this answers is not in the document]
Comment (mdebowska) [reply src/components/BetElement.tsx]: changed

File: src/components/BetCreatorForm.tsx (hunk 1 above)
Comment (mdebowska) [reply src/components/BetCreatorForm.tsx]: maybe next time

File: src/components/BetCreatorForm.tsx (hunk 2 above)
Comment (mdebowska) [reply src/components/BetCreatorForm.tsx]: Maybe next time

**Review by radek2s — APPROVED** [review]

**Pull request merged**